### PR TITLE
#603 - Fixed method name should not require a network ID in config

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: tests/core/Did.spec.ts [hunk 1]
```diff
@@ -5,11 +5,21 @@ import OperationGenerator from '../generators/OperationGenerator';
...
+        () => (Did as any).getEncodedCreateRequestFromDidString('badURL', ['anyMethodName']),
+        ErrorCode.DidInvalidDidString
+      );
+    });
+  });
+
   describe('create()', async () => {
     it('should create a short-form DID succssefully.', async () => {
-      const expectedDidMethodName = 'did:sidetree:';
+      const expectedDidMethodName = 'sidetree';
```
Comment (OR13) [inline tests/core/Did.spec.ts:20]: do you have a test for `sidetree:testnet` ?

File: lib/core/versions/latest/Did.ts [hunk 2]
```diff
@@ -63,7 +64,7 @@ export default class Did {
    */
   public static async create (didString: string, didMethodName: string): Promise<Did> {
     const methodNameParts = didMethodName.split(':');
-    if (methodNameParts.length < 2) {
+    if (methodNameParts.length > 2) {
```
Comment (isaacJChen) [inline lib/core/versions/latest/Did.ts:67]: What if method name is `sidetree:some:thing`? then we don't accept that?

Comment (thehenrytsai) [reply lib/core/versions/latest/Did.ts:67]: I'd say we don't, unless we find a need for it, for now.

Comment (OR13) [reply lib/core/versions/latest/Did.ts:67]: Thats not valid, please review did core: https://w3c.github.io/did-core/#did-syntax

Comment (OR13) [reply lib/core/versions/latest/Did.ts:67]: Apparently, this case is allowed... https://github.com/transmute-industries/vc.transmute.world/blob/master/src/services/unlockedDIDs/did:v1:test:nym:z6MkhdmzFu659ZJ4XKj31vtEDmjvsi5yDZG5L7Caz63oP39k.json

File: tests/core/Did.spec.ts (hunk 1 above)
Comment (thehenrytsai) [reply tests/core/Did.spec.ts:20]: Added a testnet DID test.

File: lib/core/versions/latest/Did.ts [hunk 3]
```diff
@@ -29,12 +30,13 @@ export default class Did {
    * @param didMethodName The expected DID method given in the DID string. The method throws SidetreeError if mismatch.
    */
   private constructor (did: string, didMethodName: string) {
-    if (!did.startsWith(didMethodName)) {
+    this.didMethodName = didMethodName;
```
Comment (logangirvin) [inline lib/core/versions/latest/Did.ts:33]:
```typescript
(did: string, public didMethodName: string) {
```

File: lib/core/versions/latest/Did.ts [hunk 4]
```diff
@@ -44,17 +46,17 @@ export default class Did {
     }
 
     if (this.isShortForm) {
-      this.uniqueSuffix = did.substring(didMethodName.length);
+      this.uniqueSuffix = did.substring(didPrefix.length);
     } else {
       // This is long-form.
-      this.uniqueSuffix = did.substring(didMethodName.length, indexOfQuestionMarkChar);
+      this.uniqueSuffix = did.substring(didPrefix.length, indexOfQuestionMarkChar);
```
Comment (logangirvin) [inline lib/core/versions/latest/Did.ts:52]: Couldn't these be combined with 43/45? It makes substring'ing on indexOfQuestionMarkChar >=0 at compile time.

File: lib/core/versions/latest/Did.ts [hunk 5]
```diff
@@ -94,35 +95,72 @@ export default class Did {
...
+    if (firstIndexOfDot === -1) {
+      throw new SidetreeError(ErrorCode.DidInitialStateValueContainsNoDot);
+    }
+
+    const lastIndexOfDot = initialState.lastIndexOf('.');
+    if (lastIndexOfDot !== firstIndexOfDot) {
+      throw new SidetreeError(ErrorCode.DidInitialStateValueContainsMoreThanOneDot);
+    }
+
+    if (firstIndexOfDot === (initialState.length - 1)) {
```
Comment (logangirvin) [inline lib/core/versions/latest/Did.ts:149]: No check for `firstIndexOfDot === 0` ?

File: tests/core/Did.spec.ts [hunk 6]
```diff
@@ -105,4 +118,49 @@ describe('DID', async () => {
...
+      const initialState = 'abc.123.'; // Intentionally having more than 1 '.'
+
+      await JasmineSidetreeErrorValidator.expectSidetreeErrorToBeThrownAsync(
+        async () => (Did as any).constructCreateOperationFromInitialState(initialState), ErrorCode.DidInitialStateValueContainsMoreThanOneDot
+      );
+      done();
+    });
+
+    it('should throw if the given initial state string has more than one dot.', async (done) => {
+      const initialState = 'abc.'; // Intentionally not having two parts after splitting by '.'
```
Comment (logangirvin) [inline tests/core/Did.spec.ts:142]: consider also testing `.abc`

File: lib/core/versions/latest/Did.ts [hunk 7]
```diff
@@ -63,24 +65,23 @@ export default class Did {
    */
   public static async create (didString: string, didMethodName: string): Promise<Did> {
     const methodNameParts = didMethodName.split(':');
-    if (methodNameParts.length < 2) {
+    if (methodNameParts.length > 2) {
```
Comment (isaacJChen) [inline lib/core/versions/latest/Did.ts:68]: Thought we want to remove this check?

File: lib/core/versions/latest/Did.ts (hunk 5 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/Did.ts:149]: Done.

File: tests/core/Did.spec.ts (hunk 6 above)
Comment (thehenrytsai) [reply tests/core/Did.spec.ts:142]: Done.

File: lib/core/versions/latest/Did.ts (hunk 7 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/Did.ts:68]: Alright, I'm getting rid off it... I got rid of it anyways in the original change.

File: lib/core/versions/latest/Did.ts (hunk 4 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/Did.ts:52]: Not sure if the ask is to combine the two if statements. If so, I'd prefer to keep them separate for readability for dummies.

File: lib/core/versions/latest/Did.ts (hunk 3 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/Did.ts:33]: Would do it if there were no other props or other props are declared and instantiated the same way. In this case prefer to keep the declaration of the `didMethodName` property explicit together with  other props for consistency.

**Pull request merged**